src/xGadgetron/cGadgetron/include/sirf/Gadgetron/trajectorypreparation.h

```diff
@@ -0,0 +1,118 @@
+/*
+SyneRBI Synergistic Image Reconstruction Framework (SIRF)
+Copyright 2020 - 2021 Physikalisch-Technische Bundesanstalt (PTB)
+
+This is software developed for the Collaborative Computational
+Project in Synergistic Reconstruction for Biomedical Imaging (formerly CCP PETMR)
+(http://www.ccpsynerbi.ac.uk/).
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+*/
+
+/*!
+\file
+\ingroup Gadgetron Extensions
+\brief Utilities for setting trajectories in ISMRMRD data.
+\author Johannes Mayer
+
+*/
+
+#ifndef TRAJECTORYPREPARATION_H
+#define TRAJECTORYPREPARATION_H
+
+#include <vector>
+
+#include <ismrmrd/xml.h>
+#include <ismrmrd/ismrmrd.h>
+
+#include "sirf/Gadgetron/gadgetron_data_containers.h"
+#include "sirf/iUtilities/LocalisedException.h"
+
+
+
+namespace sirf{
+
+/*!
+\ingroup Gadgetron Extensions
+\brief Abstract class defining the interface to set trajectories
+
+* The strategy is to perform the trajectory computation as a pre-processsing step to the
+* reconstruction. The ISMRMRD format has a 3D trajectory data field in their ISRMRMRD::Acquisition classe.
+* The interface provides set_trajectory() which populates this data field depending on the implementation.
+*/
+class TrajectoryPreparation{
+
+public:
+    TrajectoryPreparation(){}
+    virtual void set_trajectory(sirf::MRAcquisitionData& mr_acq) =0;
+    
+protected:
+
+    void update_acquisitions_info(sirf::MRAcquisitionData& mr_acq);
+
+    ISMRMRD::Encoding kspace_encoding_;
+    ISMRMRD::TrajectoryType traj_type_;
+
+    uint16_t traj_dim_;
```

Comment:
does this need to be `protected`? Or can it be `private`? (the more `private`, the better)

Reply:
inherited classes use it so I left it protected